astropy/_erfa/core.py.templ

```diff
@@ -118,46 +115,47 @@ def check_errwarn(statcodes, func_name):
         warnings.warn('ERFA function "' + func_name + '" yielded ' + wmsg, ErfaWarning)
 
 
-# <-------------------------trailing shape verification----------------------->
+# <------------------------structured dtype conversion------------------------>
 
-def check_trailing_shape(arr, shape, name):
-    try:
-        if arr.shape[-len(shape):] != shape:
-            raise Exception()
-    except:
-        raise ValueError("{0} must be of trailing dimensions {1}".format(name, shape))
+class D3Fix(numpy.ndarray):
+    # We do this to help ensure Quantity happily lets this pass...
+    def __quantity_subclass__(self, unit):
+        return type(self), True
+
+    def __array_wrap__(self, obj, context):
+        return obj.view(numpy.ndarray)
+
+
+def arrayify_inputs_and_create_d3_fix(inputs, core_dims, out_core_shape, out_dtype):
+    # Need to ensure we can get a shape
+    shapes = []
+    arrays = []
+    for input, core_dim in zip(inputs, core_dims):
+        try:
+            shape = input.shape
+        except AttributeError:
+            input = numpy.asanyarray(input)
+            shape = input.shape
+        shapes.append(shape[:-core_dim] if core_dim else shape)
+        arrays.append(input)
+
+    broadcast_shape = check_broadcast(*shapes)
+    d3_fix = numpy.empty(broadcast_shape + out_core_shape, out_dtype).view(D3Fix)
+    return arrays, d3_fix
 
-# <--------------------------Actual ERFA-wrapping code------------------------>
 
-dt_eraASTROM = numpy.dtype([('pmt','d'),
-                         ('eb','d',(3,)),
-                         ('eh','d',(3,)),
-                         ('em','d'),
-                         ('v','d',(3,)),
-                         ('bm1','d'),
-                         ('bpn','d',(3,3)),
-                         ('along','d'),
-                         ('phi','d'),
-                         ('xpl','d'),
-                         ('ypl','d'),
-                         ('sphi','d'),
-                         ('cphi','d'),
-                         ('diurab','d'),
-                         ('eral','d'),
-                         ('refa','d'),
-                         ('refb','d')], align=True)
-
-dt_eraLDBODY = numpy.dtype([('bm','d'),
-                         ('dl','d'),
-                         ('pv','d',(2,3))], align=True)
+dt_bytes1 = numpy.dtype('S1')
+dt_bytes12 = numpy.dtype('S12')
 
+# <--------------------------Actual ERFA-wrapping code------------------------>
 
 {% for constant in constants %}
 {{ constant.name }} = {{ constant.value }}
```

Comment:
The generated code here has brackets that aren't needed:

```
DR2AS = (206264.8062470963551564734)
```

This appears to be the case only for floating point values (not integers)

Reply:
The parentheses are in the `erfam.h` file. I don't think it hurts to just let them be.